dattri/algorithm/influence_function.py

```diff
@@ -606,49 +606,231 @@ def __init__(
     def transform_test_rep(
         self,
         ckpt_idx: int,
+        full_train_rep: torch.Tensor,
```

Comment:
You could cache full_train_rep as something like self.full_train_rep when calling self.cache()

Reply:
Got it.

Reply:
@jiaqima One issue is the caching of full_train_rep requires the ckpt_idx, and adding ckpt_idx would affect how attributor is used. I can make ckpt_idx to be optional though and when user does not give that parameter only loader is set and inside the attribute loop for each ckpt full_train_rep is cached once. Does that sounds reasonable?

Reply:
The total number of ckpt_idx could be determined by self.task. You could then iterate through all the ckpt_idx in self.cache() and cache them in a dictionary.

Reply:
In addition, for very large models, I think we should provide an option that estimate FIM with a subset of training data. In this case, you may want to pass an argument like the `precompute_data_ratio` in Arnoldi attributor in the __init__ of this attributor. During cache, you'll only cache subsets.

You may call it self._cached_train_reps instead of full_train_rep to make the name more accurate.

Reply:
Got it! Actually there are some potential issues in the implementation of precompute ratio in Arnoldi and I will raise that in an issue later.

Reply:
Oh in that case I can use arnoldi's sample calculation instead of what I am using now. Going to fix it soon